plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -350,27 +407,35 @@ private function set_customer_args() {
 	 * Implements filtering of orders by status.
 	 */
 	private function set_status_args() {
-		$status         = trim( sanitize_text_field( wp_unslash( $_REQUEST['status'] ?? '' ) ) );
-		$query_statuses = array();
+		$status = array_map( 'trim', (array) $this->request['status'] );
 
-		if ( empty( $status ) || 'all' === $status ) {
-			$query_statuses = array_intersect(
-				array_keys( wc_get_order_statuses() ),
-				get_post_stati( array( 'show_in_admin_all_list' => true ), 'names' )
+		if ( empty( $status ) || in_array( 'all', $status, true ) ) {
```

Comment:
note that this is never empty, because of `array_map` this will be an array of at least one element whereas earlier it would have been a string. therefore the logic inside is never executed even when the request['status'] is emtpy

Reply:
You're right. It's been fixed in d7965ac3d109c8797f7e2e62eb16261bacf43612.